packages/jest-haste-map/src/crawlers/watchman.js

```diff
@@ -104,7 +104,7 @@ module.exports = function watchmanCrawl(
 
             clocks[root] = response.clock;
             response.files.forEach(fileData => {
-              const name = root + path.sep + fileData.name;
+              const name = path.normalize(path.join(root, fileData.name));
```

Comment:
The initial implementation used to have this, however it is a source of significant slowdown (~20% of the entire time spent in jest-haste-map). Is there something else we could do here?